server/cluster_worker_test.go

```diff
@@ -307,8 +307,62 @@ func (s *mockRaftStore) handleChangePeer(c *C, req *raft_cmdpb.RaftCommandReques
 }
 
 func (s *mockRaftStore) handleSplit(c *C, req *raft_cmdpb.RaftCommandRequest) *raft_cmdpb.RaftCommandResponse {
-	// TODO later
-	return newErrorCmdResponse(errors.Errorf("unsupported request %v", req))
+	split := req.AdminRequest.Split
+	raftPeer := s.peers[req.Header.GetRegionId()]
+	splitKey := split.SplitKey
+	newRegionID := split.GetNewRegionId()
+	newPeerIDs := split.GetNewPeerIds()
+
+	region := raftPeer.region
+
+	c.Assert(newPeerIDs, HasLen, len(region.Peers))
+
+	c.Assert(string(splitKey), Greater, string(region.GetStartKey()))
+	if len(region.GetEndKey()) > 0 {
+		c.Assert(string(splitKey), Less, string(region.GetEndKey()))
+	}
+
+	newRegion := &metapb.Region{
+		RegionId: proto.Uint64(newRegionID),
+		Peers:    make([]*metapb.Peer, len(newPeerIDs)),
+		StartKey: splitKey,
+		EndKey:   append([]byte(nil), region.GetEndKey()...),
+	}
+
+	var newPeer metapb.Peer
+
+	maxPeerID := uint64(0)
+	for i, id := range newPeerIDs {
+		peer := *region.Peers[i]
+		if peer.GetStoreId() == s.storeIdent.GetStoreId() {
+			newPeer = peer
+		}
+
+		peer.PeerId = proto.Uint64(id)
+		if id > maxPeerID {
+			maxPeerID = id
+		}
+		newRegion.Peers[i] = &peer
+	}
+
+	newRegion.MaxPeerId = proto.Uint64(maxPeerID)
```

Comment:
We don't need to send max peer id. The raft group can calculate it automatically.